tasks/emoji-unicode-version-map/index.js

```diff
@@ -37,83 +10,33 @@ const aliasMap = {
 
 
 function generateEmojiUnicodeVersionMap() {
-	const addingEmojipediaUnicodeVersionPromises = Object.keys(emojipediaUnicodeVersionUrlMap).map((unicodeVersionKey) => {
-		console.log(`Working on ${unicodeVersionKey}...`);
-		const emojipediaUrl = emojipediaUnicodeVersionUrlMap[unicodeVersionKey];
-		const cacheFileName = sanitizeForFileName(emojipediaUrl);
-		const cacheFilePath = `./cached-pages/${cacheFileName}.html`;
-		// Try to read from the cache
-		return readFile(cacheFilePath, 'utf8')
-			// Fallback to fetching the page
-			.catch(() => {
-				return fetch(emojipediaUrl)
-					.then((res) => res.text())
-					.then((html) => {
-						// Save the file as a cache
-						outputFile(cacheFilePath, html);
-						return html;
-					});
-			})
-			.then((html) => {
-				scrapeEmojipediaHtml(emojiMap, html, unicodeVersionKey);
-			});
-	});
+	const emojiToUnicodeVersionMap = {};
 
-	// Save the transformed emoji map
-	const fillingZwjSequencesPromise = Promise.all(addingEmojipediaUnicodeVersionPromises)
-		.then(() => {
-			console.log('Filling out ZWJ sequences...');
-			fillZwjSequenceUnicodeVersions(emojiMap);
-		});
+	Object.keys(emojioneMap).forEach((emojioneKey) => {
+		const emoji = emojioneMap[emojioneKey];
 
-	// Setup the legacy aliases
-	const addingAliasesPromise = fillingZwjSequencesPromise
-		.then(() => {
-			Object.keys(aliasMap).forEach((aliasName) => {
-				const aliasReferenceName = aliasMap[aliasName];
-				emojiMap[aliasName] = Object.assign({}, emojiMap[aliasReferenceName]);
-			});
-		});
+		const emojiShortName = emoji.shortname;
+		const emojiNameKey = emojiShortName.slice(1, emojiShortName.length - 1);
 
-	const ensureAllPopulatedPromise = addingAliasesPromise
-		.then(() => {
-			// Verify each emoji has a `unicodeVersion`
-			let missingVersionCount = 0;
-			let notStringVersionCount = 0;
-			Object.keys(emojiMap).forEach((emojiNameKey) => {
-				const emoji = emojiMap[emojiNameKey];
-				if(emoji.unicodeVersion === undefined) {
-					console.log(`Missing unicodeVersion property on \`${emojiNameKey}\``);
-					missingVersionCount += 1;
-				}
-				if(typeof emoji.unicodeVersion !== 'string') {
-					console.log(`Invalid unicodeVersion property, must be string on \`${emojiNameKey}\``);
-					notStringVersionCount += 1;
-				}
-			});
-			if(missingVersionCount > 0) {
-				throw new Error(`Missing unicodeVersion property on ${missingVersionCount} emojis`);
-			}
-			if(notStringVersionCount > 0) {
-				throw new Error(`unicodeVersion property not a string for ${notStringVersionCount} emojis`);
-			}
-		});
+		const unicodeVersion = emoji.unicode_version.toFixed(1);
```

Comment:
What is `toFixed(1)` accomplishing here?

Perhaps there will be Unicode standards with smaller decimals in the future 🤷‍♂️

Reply:
converting a number to a string with 1 decimal

Reply:
what I could do is using `Intl.NumberFormat` to format the number with *at least* 1 decimal instead, but that would add such a dependency for an hypothetical future Unicode change.

Another solution would be to add a validation that we don't loose precision here, to make the script fail the day Unicode makes such versioning change (and not add extra dependencies until that time).
From what I can tell, it seems like Unicode even stopped using the first decimal (all recent Unicode versions are in `.0`. According to the list in the https://emojipedia.org/ footer, the last version using non-zero decimal was `6.1`.

Reply:
Implemented as a verification check, so that we know about the issue when running the build if the Unicode standards ever use multiple decimals in their versions.